<a href="https://colab.research.google.com/github/cjhaster/NRSurrogateCatalog/blob/gh-pages/_sources/events/GW170729_185629.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab"/></a>

In [1]:
%load_ext autoreload
%autoreload 2
%matplotlib inline

import pandas as pd

pd.set_option("display.max_rows", None, "display.max_columns", None)

In [2]:
import pip

try:
    __import__("nrsur_catalog")
except ImportError:
    pip.main(["install", "nrsur_catalog", "-q", ])


# GW170729_185629

Below are some plots for GW170729_185629 from the NRSurrogate Catalog.

In [3]:
from nrsur_catalog import NRsurResult

nrsur_result = NRsurResult.load("GW170729_185629", cache_dir=".nrsur_catalog_cache")
# you can specify a `cache_dir`: folder where data will be downloaded

## Summary

In [4]:
nrsur_result.summary()

Posterior 90% CI  \
Parameter                                                         
$m_1\ [M_{\odot}]$                    ${78.13}_{-8.68}^{+9.00}$   
$m_2\ [M_{\odot}]$                  ${43.16}_{-10.19}^{+10.32}$   
$\mathcal{M}\ [M_{\odot}]$            ${50.17}_{-6.69}^{+5.21}$   
$q$                                    ${0.54}_{-0.14}^{+0.19}$   
$a_1$                                  ${0.62}_{-0.37}^{+0.26}$   
$a_2$                                  ${0.50}_{-0.34}^{+0.33}$   
$\theta_1$                             ${0.88}_{-0.42}^{+0.46}$   
$\theta_2$                             ${1.38}_{-0.69}^{+0.79}$   
$\chi_{\mathrm{eff}}$                  ${0.27}_{-0.18}^{+0.15}$   
$\chi_{\mathrm{p}}$                    ${0.42}_{-0.22}^{+0.27}$   
$\alpha$                               ${5.31}_{-2.07}^{+0.47}$   
$\delta$                              ${-1.01}_{-0.22}^{+0.85}$   
$t_c\ [s]$                    ${1185389807.30}_{-0.00}^{+0.01}$   
$d_L\ [Mpc]$                    ${2328.17}_{-759.92}^{+879.42}$   
$\phi$                                 ${3.64}_{-2.43}^{+1.75}$   
$\phi$                                 ${2.50}_{-0.64}^{+0.38}$   
$\theta$                               ${1.76}_{-0.05}^{+0.04}$   
$\psi$                                 ${1.75}_{-0.91}^{+0.77}$   
$\phi_{\mathrm{JL}}$                   ${3.11}_{-2.07}^{+2.24}$   
$\phi_{12}$                            ${2.65}_{-1.73}^{+2.50}$   
$\theta_{\mathrm{JN}}$                 ${1.14}_{-0.61}^{+1.35}$   
$m_f\ [M_{\odot}]$                  ${115.76}_{-10.43}^{+9.05}$   
$\chi_f$                               ${0.78}_{-0.11}^{+0.06}$   
$v_{\rm kick}\ [{\rm km/s}]$     ${597.25}_{-396.11}^{+907.40}$   

                                                                        Prior  
Parameter                                                                      
$m_1\ [M_{\odot}]$                                                        $-$  
$m_2\ [M_{\odot}]$                                                        $-$  
$\mathcal{M}\ [M_{\odot}]$    $\text{UniformInComponentsChirpMass} [12, 400]$  
$q$                           $\text{UniformInComponentsMassRatio} [0.17, 1]$  
$a_1$                                              $\text{Uniform} [0, 0.99]$  
$a_2$                                              $\text{Uniform} [0, 0.99]$  
$\theta_1$                                            $\text{Sine} [0, 3.14]$  
$\theta_2$                                            $\text{Sine} [0, 3.14]$  
$\chi_{\mathrm{eff}}$                                                     $-$  
$\chi_{\mathrm{p}}$                                                       $-$  
$\alpha$                                                                  $-$  
$\delta$                                                                  $-$  
$t_c\ [s]$                    $\text{Uniform} [1185389807.23, 1185389807.43]$  
$d_L\ [Mpc]$                     $\text{UniformSourceFrame} [100.0, 10000.0]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\theta$                                              $\text{Sine} [0, 3.14]$  
$\psi$                                             $\text{Uniform} [0, 3.14]$  
$\phi_{\mathrm{JL}}$                               $\text{Uniform} [0, 6.28]$  
$\phi_{12}$                                        $\text{Uniform} [0, 6.28]$  
$\theta_{\mathrm{JN}}$                                $\text{Sine} [0, 3.14]$  
$m_f\ [M_{\odot}]$                                                        $-$  
$\chi_f$                                                                  $-$  
$v_{\rm kick}\ [{\rm km/s}]$                                              $-$



| Parameter                    | Posterior 90% CI                  | Prior                                           |
|:-----------------------------|:----------------------------------|:------------------------------------------------|
| $m_1\ [M_{\odot}]$           | ${78.13}_{-8.68}^{+9.00}$         | $-$                                             |
| $m_2\ [M_{\odot}]$           | ${43.16}_{-10.19}^{+10.32}$       | $-$                                             |
| $\mathcal{M}\ [M_{\odot}]$   | ${50.17}_{-6.69}^{+5.21}$         | $\text{UniformInComponentsChirpMass} [12, 400]$ |
| $q$                          | ${0.54}_{-0.14}^{+0.19}$          | $\text{UniformInComponentsMassRatio} [0.17, 1]$ |
| $a_1$                        | ${0.62}_{-0.37}^{+0.26}$          | $\text{Uniform} [0, 0.99]$                      |
| $a_2$                        | ${0.50}_{-0.34}^{+0.33}$          | $\text{Uniform} [0, 0.99]$                      |
| $\theta_1$                   | ${0.88}_{-0.42}^{+0.46}$          | $\text{Sine} [0, 3.14]$                         |
| $\theta_2$                   | ${1.38}_{-0.69}^{+0.79}$          | $\text{Sine} [0, 3.14]$                         |
| $\chi_{\mathrm{eff}}$        | ${0.27}_{-0.18}^{+0.15}$          | $-$                                             |
| $\chi_{\mathrm{p}}$          | ${0.42}_{-0.22}^{+0.27}$          | $-$                                             |
| $\alpha$                     | ${5.31}_{-2.07}^{+0.47}$          | $-$                                             |
| $\delta$                     | ${-1.01}_{-0.22}^{+0.85}$         | $-$                                             |
| $t_c\ [s]$                   | ${1185389807.30}_{-0.00}^{+0.01}$ | $\text{Uniform} [1185389807.23, 1185389807.43]$ |
| $d_L\ [Mpc]$                 | ${2328.17}_{-759.92}^{+879.42}$   | $\text{UniformSourceFrame} [100.0, 10000.0]$    |
| $\phi$                       | ${3.64}_{-2.43}^{+1.75}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi$                       | ${2.50}_{-0.64}^{+0.38}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta$                     | ${1.76}_{-0.05}^{+0.04}$          | $\text{Sine} [0, 3.14]$                         |
| $\psi$                       | ${1.75}_{-0.91}^{+0.77}$          | $\text{Uniform} [0, 3.14]$                      |
| $\phi_{\mathrm{JL}}$         | ${3.11}_{-2.07}^{+2.24}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi_{12}$                  | ${2.65}_{-1.73}^{+2.50}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta_{\mathrm{JN}}$       | ${1.14}_{-0.61}^{+1.35}$          | $\text{Sine} [0, 3.14]$                         |
| $m_f\ [M_{\odot}]$           | ${115.76}_{-10.43}^{+9.05}$       | $-$                                             |
| $\chi_f$                     | ${0.78}_{-0.11}^{+0.06}$          | $-$                                             |
| $v_{\rm kick}\ [{\rm km/s}]$ | ${597.25}_{-396.11}^{+907.40}$    | $-$                                             |


Lets make some plots!

In [5]:
# NRSurrogate corner plots

import os

param_sets = dict(
    mass=["mass_1", "mass_2", "chirp_mass", "mass_ratio"],
    spin=["a_1", "a_2", "tilt_1", "tilt_2"],
    effective_spin=["mass_ratio", "chi_eff", "chi_p"],
    sky_localisation=["luminosity_distance", "ra", "dec"],
    remnant=["final_mass", "final_spin", "final_kick"],
)
for name, params in param_sets.items():
    fname = f"GW170729_185629_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_corner(params)
        fig.savefig(fname)

    if name == "remnant":
        continue

    # LVK-Comparison plots
    fname = f"GW170729_185629_compare_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_lvk_comparison_corner(params)
        fig.savefig(fname)


## Corner Plots

### Mass


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW170729_185629_mass_corner.png"](GW170729_185629_mass_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW170729_185629_compare_mass_corner.png"](GW170729_185629_compare_mass_corner.png)
:::

::::



### Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW170729_185629_spin_corner.png"](GW170729_185629_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW170729_185629_compare_spin_corner.png"](GW170729_185629_compare_spin_corner.png)
:::

::::


### Effective Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW170729_185629_effective_spin_corner.png"](GW170729_185629_effective_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW170729_185629_compare_effective_spin_corner.png"](GW170729_185629_compare_effective_spin_corner.png)
:::

::::



### Sky-localisation


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW170729_185629_sky_localisation_corner.png"](GW170729_185629_sky_localisation_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW170729_185629_compare_sky_localisation_corner.png"](GW170729_185629_compare_sky_localisation_corner.png)
:::

::::



### Remnant

!["GW170729_185629_remnant_corner.png"](GW170729_185629_remnant_corner.png)


## Waveform posterior-predictive plot

In [6]:
fname = f"GW170729_185629_waveform.png"
if not os.path.isfile(fname):
    fig = nrsur_result.plot_signal(outdir=".")


![waveform](GW170729_185629_waveform.png)

## Animations  
Here are some animations of the posterior.

|Spin       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/F8GFaRfh10I?autoplay=1&mute=1&playlist=F8GFaRfh10I&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    |   

|Remnant       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/sR5hwRm7SJM?autoplay=1&mute=1&playlist=sR5hwRm7SJM&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    | 


## Analysis configs
Below are the configs used for the analysis of this job.

In [7]:
nrsur_result.print_configs()

Tmax:                               10000
accounting:                         None
accounting_user:                    None
adapt:                              False
allow_tape:                         True
analysis_executable:                None
analysis_executable_parser:         None
autocorr_c:                         5.0
autocorr_tol:                       50.0
bilby_zero_likelihood_mode:         False
burn_in_nact:                       50.0
calibration_model:                  CubicSpline
calibration_prior_boundary:         reflective
catch_waveform_errors:              True
channel_dict:                       {H1:DCH-CLEAN_STRAIN_C02, L1:DCH-CLEAN_STRAIN_C02, V1:Hrec_hoft_V1O2Repro2A_16384Hz}
check_point_deltaT:                 3600
clean:                              False
coherence_test:                     False
conda_env:                          None
condor_job_priority:                0
conversion_function:                None
create_summary:                     True
data

If you used this data, please [cite this work](../citation.md).

## Comments
Leave a comment below or in this [issue](https://github.com/avivajpeyi/nrsur_catlog/issues/new?title=GW170729_185629).
```{raw} html
<script src="https://utteranc.es/client.js"
        repo="avivajpeyi/nrsur_catalog"
        issue-term="GW170729_185629"
        theme="github-light"
        crossorigin="anonymous"
        async>
</script>
```